pkg/ring/model.go

```diff
@@ -442,3 +523,17 @@ func (d *Desc) RemoveTombstones(limit time.Time) {
 		}
 	}
 }
+
+// HealthCheckFunc is a function that validates whether a given token in the
+// ring is healthy and can be used.
+type HealthCheckFunc func(TokenDesc) bool
```

Comment:
This should be moved to place where it is used (=consumed) - i.e. token_navigator.go.